analysis/runtime/src/mir_loc.rs

```diff
@@ -77,9 +84,9 @@ impl From<(u64, u64)> for DefPathHash {
     }
 }
 
-impl Into<(u64, u64)> for DefPathHash {
-    fn into(self) -> (u64, u64) {
-        (self.0, self.1)
+impl From<DefPathHash> for (u64, u64) {
```

Comment:
why is this preferred?

Reply:
`From` is generally preferred over `Into` unless you can't due to orphan rules, because `From` implies `Into` due to the blanket `impl`.

See [the docs for `Into`](https://doc.rust-lang.org/std/convert/trait.Into.html):

One should avoid implementing [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) and implement [From](https://doc.rust-lang.org/std/convert/trait.From.html) instead. Implementing [From](https://doc.rust-lang.org/std/convert/trait.From.html) automatically provides one with an implementation of [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) thanks to the blanket implementation in the standard library.

Prefer using [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) over [From](https://doc.rust-lang.org/std/convert/trait.From.html) when specifying trait bounds on a generic function to ensure that types that only implement [Into](https://doc.rust-lang.org/std/convert/trait.Into.html) can be used as well.

Reply:
got it, thanks for linking all of that!